tests/robustness/validate/patch_history.go

```diff
@@ -54,28 +57,36 @@ func uniqueWatchEvents(reports []report.ClientReport) map[model.Event]traffic.Ti
 	return persisted
 }
 
-func patchOperationsWithWatchEvents(operations []porcupine.Operation, watchEvents map[model.Event]traffic.TimedWatchEvent) []porcupine.Operation {
-
+func patchOperations(operations []porcupine.Operation, watchEvents map[model.Event]traffic.TimedWatchEvent, persistedOperations map[model.EtcdOperation]int64) []porcupine.Operation {
 	newOperations := make([]porcupine.Operation, 0, len(operations))
 	lastObservedOperation := lastOperationObservedInWatch(operations, watchEvents)
 
 	for _, op := range operations {
 		request := op.Input.(model.EtcdRequest)
 		resp := op.Output.(model.MaybeEtcdResponse)
-		if resp.Error == "" || op.Call > lastObservedOperation.Call || request.Type != model.Txn {
+		if resp.Error == "" || request.Type != model.Txn {
 			// Cannot patch those requests.
 			newOperations = append(newOperations, op)
 			continue
 		}
-		event := matchWatchEvent(request.Txn, watchEvents)
-		if event != nil {
+		canMatchEvents := op.Call <= lastObservedOperation.Call
+		matchingEvent := matchWatchEvent(request.Txn, watchEvents)
```

Comment:
skipping `matchWatchEvent` when `canMatchEvents = false` could be more efficient, and make the following ` canMatchEvents && matchingEvent != nil` more concise.

Reply:
Done